client/state/stats/lists/selectors.js

```diff
@@ -119,3 +120,28 @@ export function getSiteStatsPostsCountByDay( state, siteId, query, date ) {
 	return data[ date ] || null;
 }
 
+/**
+ * Returns a parsed object of statsPublicize data for a given query, or default "empty" object
+ * if no statsStreak data has been received for that site.
+ *
+ * @param  {Object}  state    Global state tree
+ * @param  {Number}  siteId   Site ID
+ * @param  {Object}  query    Stats query object
+ * @return {Array}            Parsed Data for the query
+ */
+export const getSiteStatsParsedData = createSelector(
+	( state, siteId, statType, query ) => {
+		const data = getSiteStatsForQuery( state, siteId, statType, query );
+
+		if ( 'function' === typeof Parsers[ statType ] ) {
+			return Parsers[ statType ].call( this, data );
+		}
```

Comment:
Quite a few endpoint payloads are covered now in 
https://github.com/Automattic/wp-calypso/blob/master/client/lib/stats/stats-list/stats-parser.js - this particular one in the PR is currently not.  So moving forward we will likely see much of the same logic in the old stats parser repeated here, but in an ES6 fashion.